src/client/dfs/obj.c

```diff
@@ -671,8 +673,19 @@ dfs_obj_global2local(dfs_t *dfs, int flags, d_iov_t glob, dfs_obj_t **_obj)
 	obj->flags              = flags ? flags : obj_glob->flags;
 
 	daos_mode = get_daos_obj_mode(obj->flags);
-	rc        = daos_array_open_with_attr(dfs->coh, obj->oid, DAOS_TX_NONE, daos_mode, 1,
-					      obj_glob->chunk_size, &obj->oh, NULL);
+	if (S_ISDIR(obj->mode)) {
+		rc = daos_obj_open(dfs->coh, obj->oid, DAOS_OO_RO, &obj->oh, NULL);
```

Comment:
you cannot force RO  and should use the flags

Reply:
Shall we use "obj->flags" here?
Are there other applicable opening mode for a directory? I thought "DAOS_OO_RO" was the only valid flag to open an existing directory.

Reply:
Possible return value from get_daos_obj_mode() are DAOS_OO_RO, DAOS_OO_RW, and -1 (invalid).
Here we are handling object of an existing directory.
O_RDWR and O_WRONLY are not allow to open an existing directory in glibc open().

Reply:
you can definitely open a DFS directory with RO or RW. if you want to create a file in that dir, RO won't work. 
you are modifying the l2g g2l generic DFS code to support a directory object, so it does not matter what glibc supports or does not.

Reply:
Fixed as you suggested. Thank you!